CodeForReview.js

```diff
@@ -0,0 +1,33 @@
+var itemList = [];
+
+function addToList(itm) {
+  itemList.push(itm);
+  console.log("Item added: " + itm);
+  document.getElementById('list').innerHTML += '<li>' + itm + '</li>';
+}
+
+function checkAndRemoveFromList() {
+  var itemToRemove = document.getElementById('itemInput').value; 
+
+  if (itemList.indexOf(itemToRemove) > -1) { 
+    itemList.splice(itemList.indexOf(itemToRemove), 1); 
+    var updatedList = '';
+    for (var i = 0; i < itemList.length; i++) {
+      updatedList += '<li>' + itemList[i] + '</li>';
+    }
+    document.getElementById('list').innerHTML = updatedList;
```

Comment:
it is more efficient to first create an array with list items and then concatenate the array items into a single string after the loop. Separate DOM manipulation and reduce redundancy
```suggestion
function updateListDisplay() {
  const listElement = document.getElementById('list');
  listElement.innerHTML = itemList.map(item => '<li>' + item + '</li>').join('');
}
```